app/controllers/api/proxy.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+use Appwrite\Event\Certificate;
+use Appwrite\Event\Delete;
+use Appwrite\Event\Event;
+use Appwrite\Extend\Exception;
+use Appwrite\Network\Validator\CNAME;
+use Appwrite\Utopia\Database\Validator\Queries\Rules;
+use Appwrite\Utopia\Response;
+use Utopia\App;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Helpers\ID;
+use Utopia\Database\Query;
+use Utopia\Database\Validator\UID;
+use Utopia\Domains\Domain;
+use Utopia\Validator\Domain as ValidatorDomain;
+use Utopia\Validator\Text;
+use Utopia\Validator\WhiteList;
+
+App::post('/v1/proxy/rules')
+    ->groups(['api', 'proxy'])
+    ->desc('Create Rule')
+    ->label('scope', 'rules.write')
+    ->label('event', 'rules.[ruleId].create')
+    ->label('audits.event', 'rule.create')
+    ->label('audits.resource', 'rule/{response.$id}')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_ADMIN])
+    ->label('sdk.namespace', 'proxy')
+    ->label('sdk.method', 'createRule')
+    ->label('sdk.description', '/docs/references/proxy/create-rule.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_PROXY_RULE)
+    ->param('domain', null, new ValidatorDomain(), 'Domain name.')
+    ->param('resourceType', null, new WhiteList(['api', 'function']), 'Action definition for the rule. Possible values are "api", "function", or "redirect"')
+    ->param('resourceId', '', new UID(), 'ID of resource for the action type. If resourceType is "api" or "url", leave empty. If resourceType is "function", provide ID of the function.', true)
```

Comment:
url isn't a valid resourceType, right?

```suggestion
    ->param('resourceId', '', new UID(), 'ID of resource for the action type. If resourceType is "api", leave empty. If resourceType is "function", provide ID of the function.', true)
```

Reply:
Also, is redirect a valid value for resourceType?

Reply:
Its not. Fixed both